internal/graph/builder.go

```diff
@@ -0,0 +1,101 @@
+package graph
```

Comment:
Can we please put this under a graph specific package. The vision is that we should eventually have (I wish we had started this way) `internal/graph/check` and `internal/graph/reverseexpand` etc.. Each query pattern ontop of the graph model warrants its own package. CheckResolverBuilder belongs to the internal (Check specific) graph API.

Reply:
Agreed, but that would mean the whole of the `graph` package which doesn't have anything else other than check logics should be in this folder, Let me make this change towards the end of the PR to keep it neater for diff. Ideally I would do this in another PR related to restructuring but let me do it towards the end of this PR.